### PR TITLE
Add clang-format test

### PR DESCRIPTION
**Pull request by wershlak**
Adding a clang-format test.

I formatted one file as an example. We can change the config to suit our needs and I can format all the files once we agree on the right formatting.

NOTE: unlike the uncrustify check this check is only checking the git diff so only LINES which are updated are checked. IE. this is safe to merge.

### CODE REVIEW

File: code/code/cmd/cmd_attribute.cc [hunk 1]
```diff
@@ -496,36 +541,36 @@ const char *statBonusDescr(int percent)
...
-    return;
+  if (!desc) return;
 
   one_argument(arg, cmdbuf);
   if (cmdbuf.empty()) {
-    sendTo("Syntax: attribute <personal | condition | statistics | reset>\n\r");
+    sendTo(
+        "Syntax: attribute <personal | condition | statistics | "
+        "reset>\n\r");
     return;
```
Comment (drewhershey) [inline code/code/cmd/cmd_attribute.cc:560]: This is kinda weird looking - wonder what rule causes it?

Comment (wershlak) [reply code/code/cmd/cmd_attribute.cc]: that's ugly. Added `BreakStringLiterals: false`

**Comment by wershlak** [comment]
@cizra Any opinion on what style we use? Also do you think it's a good idea to format the entire repo once we decide on a style?

**Comment by cizra** [comment]
> @cizra Any opinion on what style we use? Also do you think it's a good idea to format the entire repo once we decide on a style?

I'm 100% sure we should format the whole repo in one go. See https://www.moxio.com/blog/43/ignoring-bulk-change-commits-with-git-blame how to make this commit not end up in `git blame`.

As for which style to use - I haven't read them all. Based on cursory glance, it seems Google / LLVM / Chromium are all similar enough.
In particular, I like:
- tabs
- Egyptian braces
- curly elses
- ... and generally to preserve vertical space

**Review by cizra — CHANGES_REQUESTED** [review]

**Comment by drewhershey** [comment]
Going to close this out, as I have a .clang-format file I've been tweaking for a while now that I'm going to use for this purpose on the entire codebase at once, which will be done in a new PR.

**Pull request closed without merging**